core/templates/services/debouncer.service.ts

```diff
@@ -27,32 +27,28 @@ export class DebouncerService {
   // be invoked. The function only gets executed after it stops being called
   // for `wait` milliseconds.
   debounce(func: () => void, millisecsToWait: number): () => void {
-    let timeout;
-    let context = this;
-    let args = arguments;
-    let timestamp;
-    let result;
+    let timeout: number;
+    let context: this | null = this;
+    let timestamp: number;
+    let result: void;
 
     const later = () => {
       let last = new Date().getTime() - timestamp;
       if (last < millisecsToWait) {
-        timeout = setTimeout(later, millisecsToWait - last);
+        timeout = window.setTimeout(later, millisecsToWait - last);
```

Comment:
Use windoRef instead.

Reply:
Sorry I didn't get it. Can You please elaborate? Do I have to use ```windoRef.settimeout(...)``` or it is something else?

Reply:
> Use windoRef instead.

If I am using windowRef instead of window then it is showing ```Property 'setTimeout' does not exist on type 'WindowRef'.```.

Reply:
I think you'll have to do WindowRef.nativeWindow.setTimeout (not sure!) Please check how WindowRef is used in other places in the codebase.

Reply:
Done